### PR TITLE
Pipeline

### PR DESCRIPTION
**Pull request by VinaiRachakonda**
Basic pipeline features.

### CODE REVIEW

File: client/components/Company/CardView/CardView.tsx [hunk 1]
```diff
@@ -1,49 +1,49 @@
-import React from "react";
-import styled from "styled-components";
-import { Draggable } from "react-beautiful-dnd";
-import { Card, Button } from "antd";
-import { Link } from "react-router-dom";
+import React from 'react';
+import Link from 'next/link';
```
Comment (casey-chow) [inline client/components/Company/CardView/CardView.tsx]: Should be using the Link imported from routes.ts not this one.

File: client/components/Pipeline/Column.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+`;
+
+interface ColumnProps {
+  title: any;
+  id: any;
+  companies: any;
+}
+
+interface ColumnState {
+  companies: any;
```
Comment (casey-chow) [inline client/components/Pipeline/Column.tsx]: `any[]`

File: client/components/Pipeline/Column.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+export default class Column extends React.Component<ColumnProps, ColumnState> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      companies: props.companies,
```
Comment (casey-chow) [inline client/components/Pipeline/Column.tsx:47]: This is a good candidate to try hooks if you want

File: client/components/Pipeline/IndividualButton.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import Button from 'react-bootstrap/lib/Button';
+import { Subscribe } from 'unstated';
+import PipelineContainer from '../../containers/PipelineContainer';
+
+interface DropdownProps {
+  loggedInPartnerName: string;
+}
+
+export default class IndividualButton extends PureComponent<DropdownProps> {
+  render() {
```
Comment (casey-chow) [inline client/components/Pipeline/IndividualButton.tsx]: Prefer function component

File: client/components/Pipeline/Kanban.tsx [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+import Form from 'react-jsonschema-form';
+import axios from 'axios';
+import CustomDropdown from './Dropdown';
+import GroupButton from './GroupButton';
+import IndividualButton from './IndividualButton';
+import Column from './Column';
+import transformData from './utils';
+import schema from '../../shared/schema';
+
+const cschema = schema.companies;
```
Comment (casey-chow) [inline client/components/Pipeline/Kanban.tsx]: `companySchema`

File: client/components/Pipeline/Kanban.tsx [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+    loading: true,
+    columns: {},
+    columnOrder: [],
+    partnerNames: new Set([]),
+    visible: false,
+  };
+
+  componentDidMount() {
+    axios
+      .get('http://localhost:3000/api/companies?archived=false', axiosConfig)
```
Comment (casey-chow) [inline client/components/Pipeline/Kanban.tsx:68]: Can we move this to the feathers client? It’s gonna be a drag in production

File: client/components/Pipeline/utils.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
+/*
```
Comment (casey-chow) [inline client/components/Pipeline/utils.tsx]: Can we move this to `client/lib`?

File: client/next.config.js [hunk 8]
```diff
@@ -1,39 +1,42 @@
 const config = require('config');
 const path = require('path');
 const withTypescript = require('@zeit/next-typescript');
+const withCSS = require('@zeit/next-css');
```
Comment (casey-chow) [inline client/next.config.js]: Life might be easier if you compose using `_.flow` instead.

File: client/pages/pipeline.tsx [hunk 9]
```diff
@@ -1,171 +1,27 @@
...
-  display: flex;
-`;
-
-interface PipelineProps {
-  state: any,
-}
-
-interface PipelineState {
-  loading: boolean,
-  columns: Object,
```
Comment (casey-chow) [inline client/pages/pipeline.tsx:20]: Lowercase object, common TypeScript error.

File: client/components/Pipeline/Kanban.tsx [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+const AppContainer = styled.div`
+  display: flex;
+`;
+
+const axiosConfig = {
+  headers: { 'Access-Control-Allow-Origin': '*' },
+};
+
+interface KanbanProps {
+  userc: any;
```
Comment (casey-chow) [inline client/components/Pipeline/Kanban.tsx]: Unclear prop name

File: server/services/companies/companies.hooks.ts [hunk 11]
```diff
@@ -46,18 +46,42 @@ const votedPartners = {
   },
 };
 
+const pointPartners = {
```
Comment (casey-chow) [inline server/services/companies/companies.hooks.ts:49]: We can also use Objection to do the join. Any reason not to here?

**Review by casey-chow — CHANGES_REQUESTED** [review]

File: client/components/Pipeline/Column.tsx (hunk 3 above)
Comment (VinaiRachakonda) [reply client/components/Pipeline/Column.tsx:47]: Hmmm maybe later haha

File: server/services/companies/companies.hooks.ts (hunk 11 above)
Comment (VinaiRachakonda) [reply server/services/companies/companies.hooks.ts:49]: Do you mean $eager loading? I thought that was between tables. I'm just grabbing User info from an array of userIds

**Review by casey-chow — APPROVED** [review]

**Pull request merged**